### ISSUE TITLE
Fenders connected to wheels do not consistently stay on top of the wheel

### ISSUE THREAD

**Issue by GreatEmerald** [GreatEmerald/UT3Vehicles#109]
(@HellDragon-HK: This is example of how to create an issue on a bug that affects multiple vehicles.)

The Hellbender, SPMA and Leviathan have fenders that are connected through bones to the wheels. In UT2004, the wheel bone and all its attachments rotate when the vehicle moves, making the vehicles appear as if they're constantly running over their own fenders.

At the moment as a workaround, the location of the fenders is updated every tick to where it normally should be on the roll axis. However, this does not seem to be a reliable solution, possibly because vehicles get updated at double the tick rate, making the updates to the fender unsmooth. In addition, it causes extra work every tick and may cause replication issues and issues when a driver has exited the vehicle. So a better solution would be appreciated.

Is there a way to somehow tell the game to "freeze" a bone in place? Ideally only on a particular rotation axis, since we'd want the fenders to follow the wheels when they are turned to the sides. An alternative solution would be to edit the mesh to make the fender bones not connected to the wheels, and update their rotation as needed instead (which would be far less noticeable).

**Comment by HellDragon-HK** [comment]
A little confusing to make it all instead of one at a time to me because just because a fix will work for one doesn't mean the same fix will work for others when found but who knows, I can agree that the fenders are a multiple issue type.

The stock Paladin has fenders as part of it's wheel bones by the look of it in the UT2004 Editor and they don't turn with the wheel, might be a good place to start?

**Comment by GreatEmerald** [comment]
Hm, the only thing I can see in its code that might relate to it is `SteeringBone*`. Could it be that on the Paladin the fenders are actually attached to this steering bone rather than to the wheels?

**Comment by HellDragon-HK** [comment]
To me it looks like the wheels and fenders are the same bone, but I can get you a quick couple of screens of the editor so you can see for yourself since you don't have it available to you at this time if you like

**Comment by GreatEmerald** [comment]
Perhaps. It could be that both bones are at the same place, yet not connected to the same parent bone.

**Comment by HellDragon-HK** [comment]
![ss_001](https://user-images.githubusercontent.com/34013048/34424632-b6149c06-ebea-11e7-9ce9-3b8bfa8c7bb0.jpg)
![ss_002](https://user-images.githubusercontent.com/34013048/34424636-bb152d42-ebea-11e7-9901-94f513b8f378.jpg)

If you don't ever want me to try screens directly like this again please say so, thought it would be useful if everyone involved could easily see though.

**Comment by 100GPing100** [comment]
If I remember correctly I think I edited the UT3 Paladin mesh before importing it into the editor to be able to have a bone just for the tire.

**Comment by GreatEmerald** [comment]
@HellDragon-HK It's fine to upload images here. How does this look in the bone view, without displaying the vehicle mesh? And where is the `SteeringWheel` bone, what is it connected to?

**Comment by HellDragon-HK** [comment]
Hold on a second I'll get a shot of it if I can

**Comment by HellDragon-HK** [comment]
Uh there isn't a ````SteeringBone?

![ss_001](https://user-images.githubusercontent.com/34013048/34457327-1ad44284-ed73-11e7-82ed-bc094c8c525e.jpg)
![ss_002](https://user-images.githubusercontent.com/34013048/34457328-1ceef7da-ed73-11e7-9ecb-a4b8700c93b2.jpg)

**Comment by 100GPing100** [comment]
I'm betting `SteeringWheel` is a root bone, right where all those mangled names are.

**Comment by HellDragon-HK** [comment]
Is there anyway to make them easier to read, I think I can see Wheel in there at the center but it look like it's near the first of the word which doesn't make sense...WheelAttach?

**Comment by 100GPing100** [comment]
Only way to see all the bones properly and which vertices of the mesh are attached to them is by exporting the mesh; installing blender and the ActorX plugins; opening up the mesh in blender. Other than that there is no way to see it all properly, which is a shame to be honest.

**Comment by HellDragon-HK** [comment]
What's your thought on how trying to add a collision box treats it as if it doesn't exist?

**Comment by 100GPing100** [comment]
Well I haven't done that in a long time; from what I remember adding a collision box requires you to also give it a bone name to attach to. If you look at the Viper it has a bunch of collision boxes, and the ones on the wings move with the wings because they are attached to a bone (I assume from how skeletal meshes work, otherwise I see no way of how the box would move with the wing).

**Comment by HellDragon-HK** [comment]
Right, it won't attach to the name of ````SteeringWheel```` but will with ````CannonAttach```` or ````8WheelerBase```` leading me to believe it doesn't exist and I had to radius set so it was easy to see when it showed

**Comment by 100GPing100** [comment]
Then my guess would be that the bone `SteeringWheel` does not exist and that the engine defaults to the root bone when this happens.

**Comment by 100GPing100** [comment]
Here are all the bones in `ONSBPAnimations.ShockTankMesh`:
![image](https://user-images.githubusercontent.com/5787775/34458684-76aff444-edd3-11e7-8c7b-50ad2cb5d23b.png)

As we can see there is no `SteeringWheel` bone, thus the engine must be using, by default, the root bone, which in this case is `8WheelerBase`.

**Comment by GreatEmerald** [comment]
Heh, nice Oxygen cursor there.

But this doesn't get us any closer to figuring out why the fenders of the Paladin turn with the wheels... `ONSShockTank` is also not native, so whatever causes them to turn must be there somewhere. Could it be that the fenders are connected to the suspension bones?

**Comment by 100GPing100** [comment]
Let me just figure out again how editing bones works so I can see to which bone the fender belongs to. Your assumption is too the same one I've been having.

**Comment by 100GPing100** [comment]
Here is the answer, in `ONSShockTank` the fenders belong to the suspension bone.

In the following screenshot I moved the bone `8WheelerWheel03` down and the bone `Suspension_Right2` up:
![image](https://user-images.githubusercontent.com/5787775/34462931-d5d7399c-ee45-11e7-8a2f-60e3eed1d01a.png)

As can be seen, the fender moved up with the suspension bone, and only the wheel moved down with the wheel bone; with this we can conclude that the fender is part of the suspension.

**Comment by HellDragon-HK** [comment]
So to fix the fender problem they need to be connected to the suspensions then, can that be done in code or is it another binary change that's needed?

**Comment by 100GPing100** [comment]
It requires the mesh to be exported, then edited using 3D modeling software, exported back into psk and then needs to be imported back into Unreal Ed again and all the collision needs to be re-done again, most likely.

**Comment by HellDragon-HK** [comment]
Ouch which vehicles will this affect, list them and I'll write down the collision boxes for them
